lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -173,8 +187,14 @@ where
     fn decode<B: ReadIoBuffer>(&mut self, buf: &mut B, events: &mut EventBuffer) -> Result<usize, Self::Error> {
         let data = buf.chunk();
 
+        let message_type = parse_message_type(data);
+        if message_type == MessageType::Event {
+            return self.decode_event(buf, events);
+        }
+
         // Decode the payload and get the representative parts of the metric.
-        let (remaining, (metric_name, tags_iter, values_iter, mut metadata)) = parse_dogstatsd(data, &self.config)?;
+        let (remaining, (metric_name, tags_iter, values_iter, mut metadata)) =
+            parse_dogstatsd_metric(data, &self.config)?;
```

Comment:
I think i did try that but I had to put `decode_metric` inside the `impl<TMI> DogstatsdCodec<TMI>` block (since I can't put it inside `impl<TMI> Decoder for DogstatsdCodec<TMI>` due to the trait I believe) and then I think I ran into some error with the new tag filterer stuff `        let filtered_tags_iter = TagFilterer::new(tags_iter.clone(), &self.tag_metadata_interceptor);` but I'll take a look again in the service checks PR.

Reply:
Hmm, well, we can leave this for the service checks PR or even a follow-on PR after that, just to make the diff easier to review as we play around with it.